openaz-xacml/src/main/java/org/apache/openaz/xacml/std/jaxp/JaxpRequest.java

```diff
@@ -114,19 +115,17 @@ public static JaxpRequest load(File fileXmlRequest) throws ParserConfigurationEx
             logger.error("No Document returned parsing \"" + fileXmlRequest.getAbsolutePath() + "\"");
             return null;
         }
-
-        NodeList nodeListRoot = document.getChildNodes();
-        if (nodeListRoot == null || nodeListRoot.getLength() == 0) {
-            logger.warn("No child elements of the XML document");
-            return null;
-        }
-        Node nodeRoot = nodeListRoot.item(0);
-        if (nodeRoot == null || nodeRoot.getNodeType() != Node.ELEMENT_NODE) {
-            logger.warn("Root of the document is not an ELEMENT");
-            return null;
+        
+        Node nodeRoot = document.getDocumentElement();
+        
+        if(context == null) {
+            synchronized (JaxpRequest.class) {
```

Comment:
Now you are breaking DRY principle, again -1 for that. Thing, again to make it better, separate class is always a good option.